blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGeneratorManager.java

```diff
@@ -64,33 +51,24 @@ public void requestCodeGeneration(List<String> blockDefinitionsJsonPaths,
                                       CodeGeneratorCallback codeGenerationCallback) {
 
         if (workspace.hasBlocks()) {
-            if (isBound()) {
-                try {
-                    final StringOutputStream serialized = new StringOutputStream();
-                    workspace.serializeToXml(serialized);
+            try {
+                final StringOutputStream serialized = new StringOutputStream();
+                workspace.serializeToXml(serialized);
+                CodeGenerationRequest codeGenerationRequest =
+                    new CodeGenerationRequest(
+                        serialized.toString(),
+                        codeGenerationCallback,
+                        blockDefinitionsJsonPaths,
+                        generatorsJsPaths);
+                mCodeGenerationConnection.requestCodeGeneration(codeGenerationRequest);
```

Comment:
I went with 1, and discovered that it removes all the logic from CodeGenerationManager, so I've done that, then relocated the lazy loading into CodeGenerationManager and deleted the new LazyServiceConnection class.